### PR TITLE
Docs: unified organisation | CI/CD

### PR DESCRIPTION
**Pull request by Mathias-Boulay**
[Draft moment]

### CODE REVIEW

File: Architecture.md [hunk 1]
```diff
@@ -36,17 +43,30 @@
...
+Each repository has a GitHub Actions CI to execute format check, linter and tests at each push. This ensures a good code quality and a good test coverage for our entire project.
+
+Most projects use our own Actions available in a [central repository](https://github.com/paastech-cloud/.github). 
+This ensures that all projects have the same CI/CD pipeline and that the CI/CD pipeline is easily maintainable.
+Moreover, thanks to Rust easy documentation creation process, a simple CI job is able to build the documentation and deploy it on [GitHub Pages](https://pages.github.com/) for every Rust project.
+Furthermore, at a tag push on main, or on a workflow call, a Docker image is build for any project. 
+This image is then pushed as a repository package to the [GitHub Container Registry](https://ghrc.io) and is available for use.
 
+On top of that, the web UI is automatically deployed on [Github Pages](https://pages.github.com/).
+
```
Comment (BioTheWolff) [inline Architecture.md:55]:
```suggestion

#### Deployment
```

File: Architecture.md [hunk 2]
```diff
@@ -36,17 +43,30 @@
 
 #### CI/CD
 
-***TODO: [UNIFIED WORK] describe use of Github Actions as means of CI/CD***
+Each repository has a GitHub Actions CI to execute format check, linter and tests at each push. This ensures a good code quality and a good test coverage for our entire project.
+
+Most projects use our own Actions available in a [central repository](https://github.com/paastech-cloud/.github). 
+This ensures that all projects have the same CI/CD pipeline and that the CI/CD pipeline is easily maintainable.
+Moreover, thanks to Rust easy documentation creation process, a simple CI job is able to build the documentation and deploy it on [GitHub Pages](https://pages.github.com/) for every Rust project.
+Furthermore, at a tag push on main, or on a workflow call, a Docker image is build for any project.
```
Comment (BioTheWolff) [inline Architecture.md:51]:
```suggestion
Furthermore, at a tag push on main, or on a workflow call, a Docker image is built for every project that has this workflow set up.
```

File: Architecture.md [hunk 3]
```diff
@@ -36,17 +43,30 @@
...
+Furthermore, at a tag push on main, or on a workflow call, a Docker image is build for any project. 
+This image is then pushed as a repository package to the [GitHub Container Registry](https://ghrc.io) and is available for use.
 
+On top of that, the web UI is automatically deployed on [Github Pages](https://pages.github.com/).
+
+To deploy every application of PaaSTech in a single machine, a `docker-compose.yml` was made alongside a `setup.sh` script to set up the host machine easily and deploy the entire project. 
+They are both available in [the `infra` repository](https://github.com/paastech-cloud/infra).
+In this `docker-compose.yml`, we create 2 networks, one for the internal applications, and one for the client applications.
+Each application is then deployed in a container, with environment variables that can be set up by the user in the `.env` file, and volumes as needed.
+This file also ensure that the API is actually available at [https://api.paastech.cloud](https://api.paastech.cloud), that the git controller is available on the selected port, that every container are started in order, can communicate and that Traefik is set up correctly.
```
Comment (BioTheWolff) [inline Architecture.md:60]:
```suggestion
This file also ensure that:
- the API is actually available at [https://api.paastech.cloud](https://api.paastech.cloud)
- the git controller is available on the selected port
- all containers are started in order
- all containers can communicate with eachother
- Traefik is set up correctly.
```

**Review by BioTheWolff — CHANGES_REQUESTED** [review]

File: Architecture.md [hunk 4]
```diff
@@ -79,4 +105,13 @@
 
 ### Organisational overview
 
-***TODO: [UNIFIED WORK] from an organisation standpoint, how was the entire team organised, how did the squads interact***
+In the beginning, we held a few meetings including everyone to specify all of the necessary functionalities. Due to high number of people we weren't really able to be as productive as we wanted but managed to create an overall specifications chart.
```
Comment (BioTheWolff) [inline Architecture.md:111]:
```suggestion
In the beginning, we held a few meetings inviting everyone to specify all of the necessary functionalities. As a result of the large number of participants, we weren't really able to be as productive as we wanted but managed to create a first baseline of a specifications chart.
```

File: Architecture.md [hunk 5]
```diff
@@ -36,17 +43,36 @@
...
+
+Most projects use our own Actions available in a [central repository](https://github.com/paastech-cloud/.github). 
+This ensures that all projects have the same CI/CD pipeline and that the CI/CD pipeline is easily maintainable.
+Moreover, thanks to Rust easy documentation creation process, a simple CI job is able to build the documentation and deploy it on [GitHub Pages](https://pages.github.com/) for every Rust project.
+Furthermore, at a tag push on main, or on a workflow call, a Docker image is built for every project that has this workflow set up.
+This image is then pushed as a repository package to the [GitHub Container Registry](https://ghrc.io) and is available for use.
 
+On top of that, the web UI is automatically deployed on [Github Pages](https://pages.github.com/).
+
+#### Deployment
```
Comment (BioTheWolff) [inline Architecture.md:56]: Add a line return between the title and the paragraph

**Review by BioTheWolff — CHANGES_REQUESTED** [review]

File: Architecture.md (hunk 5 above)
Comment (remi-espie) [reply Architecture.md:56]: isn't it there already ?

File: Architecture.md [hunk 6]
```diff
@@ -36,17 +43,39 @@
 
 #### CI/CD
 
-***TODO: [UNIFIED WORK] describe use of Github Actions as means of CI/CD***
+Each repository has a GitHub Actions CI to execute format check, linter and tests at each push. This ensures a good code quality and a good test coverage for our entire project.
+
+Most projects use our own Actions available in a [central repository](https://github.com/paastech-cloud/.github). 
+This ensures that all projects have the same CI/CD pipeline and that the CI/CD pipeline is easily maintainable.
+Moreover, thanks to Rust easy documentation creation process, a simple CI job is able to build the documentation and deploy it on [GitHub Pages](https://pages.github.com/) for every Rust project.
+Furthermore, at a tag push on main, or on a workflow call, a Docker image is built for every project that has this workflow set up.
```
Comment (MurielParaire) [inline Architecture.md:51]: J'aime pas trop la tournure du début de la phrase
Do you want to say every tag by 'at a tag' or a specific tag ?

File: Architecture.md [hunk 7]
```diff
@@ -36,17 +43,39 @@
...
 
+#### Deployment
+
+To deploy every application of PaaSTech on a single machine, a `docker-compose.yml` was made alongside a `setup.sh` script to set up the host machine easily and deploy the entire project.
+As of today, PaaSTech is not scalable and is not meant to be deployed on multiple machines. 
+Furthermore, even if the setup script does a bit of configuration, some folder are still needed to be created manually, the `.env` file needs to be filled and a [custom script](https://github.com/paastech-cloud/git-repo-manager/blob/main/hooks/post-receive) must be added in a specific directory on the host machine (`/srv/hooks` by default). 
+This deployment method is available in [PaaSTech' `infra` repository](https://github.com/paastech-cloud/infra).
+In this `docker-compose.yml`, we create 2 networks, one for the internal applications, and one for the client applications.
+Each application is then deployed in a container, with environment variables that can be set up by the user in the `.env` file, and volumes as needed.
+This file also ensure that:
```
Comment (MurielParaire) [inline Architecture.md:64]: I believe ensure takes an s here
```suggestion
This file also ensures that:
```

File: Architecture.md [hunk 8]
```diff
@@ -36,17 +43,39 @@
...
+- the API is actually available at [https://api.paastech.cloud](https://api.paastech.cloud)
+- the git controller is available on the selected port
+- all containers are started in order
+- all containers can communicate with eachother
```
Comment (MurielParaire) [inline Architecture.md:68]: I think they're two words
```suggestion
- all containers can communicate with each other
```

File: Architecture.md [hunk 9]
```diff
@@ -36,17 +43,39 @@
...
 ### Definitions
 
-- the Service refers to PaaSTech as a whole;
-- a Client is a user account created by an end user against the Service;
-- a Project is a materialisation of a Git repository, created by a Client using either the web frontend or the CLI. A Project can be deployed by the Client by pushing its code to the Service.
-- an Application (also referred to as Deployment) is an atomic unit of code, and is the result of a Project deployment. This unit is internally managed and can only be configured to a certain extent by the Client.
+- The Service refers to PaaSTech as a whole;
+- A Client is a user account created by an end user against the Service;
+- A Project is a materialisation of a Git repository, created by a Client using either the web frontend or the CLI. A Project can be deployed by the Client by pushing its code to the Service.
+- An Application (also referred to as Deployment) is an atomic unit of code, and is the result of a Project deployment. This unit is internally managed and can only be configured to a certain extent by the Client.
```
Comment (MurielParaire) [inline Architecture.md:78]:
```suggestion
- An Application (also referred to as Deployment) is an atomic unit of code and is the result of a Project deployment. This unit is internally managed and can only be configured to a certain extent by the Client.
```

**Review by MurielParaire — CHANGES_REQUESTED** [review]

File: Architecture.md (hunk 8 above)
Comment (BioTheWolff) [reply Architecture.md:188]: Nope, one word only!

**Review by MurielParaire — APPROVED** [review]

**Review by BioTheWolff — APPROVED** [review]

**Pull request merged**